### PR TITLE
Support swapping to disk for large nested archives

### LINKED ISSUE #40: Support inspection of large archives in memory-constrained scenarios
In the current implementation, the detection logic for detecting vulnerabilities in nested ZIP archives requires the entire nested ZIP to be loaded into memory. If this program is run in low-memory scenarios with large nested archives, this could pose an issue.

In order to mitigate this, the implementation should either be updated such that entire nested archives are not loaded into memory or so that the nested archives (or the relevant files within them) are written out to disk and examined there so that disk is used rather than memory.

### PR DESCRIPTION
**Pull request by glynternet**
Closes https://github.com/palantir/log4j-sniffer/issues/40

Adds a feature that can be enabled with by setting `--nested-archive-disk-swap-max-size` to a positive non-zero value.

When a nested zip file is encountered that is above the `--nested-archive-max-size`, space will be used on disk to write out the archive temporarily to be able to inspected without incurring a large memory penalty.
Deeply nested archives can be temporarily written out to disk, even if the parent archives are written to disk, providing that the total space taken up is less than or equal to `--nested-archive-disk-swap-max-size`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -249,39 +249,43 @@ Usage:
...
+                                                             Patterns should be relative to the provided root.
+                                                             e.g. ignore "^/proc" to ignore "/proc" when using a crawl root of "/"
+      --json                                                 If true, output will be in JSON format
+      --maximum-average-obfuscated-class-name-length int     The maximum class name length for a class to be considered obfuscated. (default 3)
+      --maximum-average-obfuscated-package-name-length int   The maximum average package name length a class to be considered obfuscated. (default 3)
+      --nested-archive-disk-swap-dir string                  When nested-archive-disk-swap-max-size is non-zero, this is the directory in which temporary files will be created for writing temporary large nested archives to disk. (default "/tmp")
+      --nested-archive-disk-swap-max-size uint               The maximum size in bytes of disk space allowed to use for inspecting nest archives that are over the nested-archive-max-size.
```
Comment (hpryce) [inline README.md:277]: We should say what the default is here

Comment (hpryce) [reply README.md:277]: (And in the actual implementation too of course)

Comment (glynternet) [reply README.md:277]: Added in https://github.com/palantir/log4j-sniffer/pull/98/commits/75761ecab605535572e6dc89d5009c4c1ab36b46

I couldn't quite find where in the implementation to add it as the default value of the exported struct field is 0, where the current comment is true in that 0 is a valid maximum number of bytes to be used. Let me know if I'm misunderstanding here and you feel that is still valuable to document, though :)

Comment (hpryce) [reply README.md:277]: Sorry, I meant in the text to document the default being 0 i.e. disabled

**Review by hpryce — APPROVED** [review]

**Pull request merged**